general/g.tempfile/g.tempfile.html

```diff
@@ -36,12 +47,30 @@ <h2>NOTES</h2>
 creates a different (i.e. unique) name.
 
 Although GRASS does eventually get around to removing
-tempfiles that have been left behind, the programmer should
-make every effort to remove these files. They often get
-large and take up disk space. If you write /bin/sh scripts,
-learn to use the /bin/sh <em>trap</em> command. If you
-write /bin/csh scripts, learn to use the /bin/csh
-<em>onintr</em> command.
+temporary files and directories that have been left behind,
+the programmer should make every effort to remove these files.
+They often get large and take up disk space. If you write /bin/sh
+scripts, learn to use the /bin/sh <em>trap</em> command. If you
+write /bin/csh scripts, learn to use the /bin/csh <em>onintr</em>
+command. In Python, use an <em>atexit</em> procedure.
+
+<p>
+If the size of temporary files is not an issue, it is recommended
+to use <i>NamedTemporaryFile</i> with a context manager to create a
+temporary file in Python.<br>
```

Comment:
Just to be clear, you can pass _dir_ to both NamedTemporaryFile and TemporaryDirectory, so you can (could) use the 100% way with the project-based tmp dir if you get the project-mapset-hostname path in some way even without creating subdirectories there.